docs/rules/no-extra-parens.md

```diff
@@ -9,6 +9,8 @@ This rule always ignores extra parentheses around the following:
 * RegExp literals such as `(/abc/).test(var)` to avoid conflicts with the [wrap-regex](wrap-regex.md) rule
 * immediately-invoked function expressions (also known as IIFEs) such as `var x = (function () {})();` and `((function foo() {return 1;})())` to avoid conflicts with the [wrap-iife](wrap-iife.md) rule
 
+In addition, this rule ignores parentheses around arrow function arguments to avoid conflicts with the [arrow-parens](arrow-parens.md) rule.
```

Comment:
Hmm... The only reason I did that was because arrow function arguments are not an expression with malleable precedence, unlike RegExp literals and IIFEs, so it felt like apples/oranges to me. But I don't have a strong preference either way.